master/internal/api_experiment.go

```diff
@@ -92,6 +92,33 @@ func (a *apiServer) GetExperiments(
 		if len(req.Users) != 0 && !found {
 			return false
 		}
+
+		if len(req.Labels) != 0 {
+			found = true
+
+			if len(v.Labels) == 0 {
+				return false
+			}
+
+			for _, reqLabel := range req.Labels {
+				hasAMatch := false
+				for _, expLabel := range v.Labels {
+					if hasLabel := strings.Contains(strings.ToLower(expLabel), strings.ToLower(reqLabel)); hasLabel {
+						hasAMatch = true
+						break
+					}
+				}
+				if !hasAMatch {
+					found = false
+					break
+				}
+			}
```

Comment:
Non-blocking: would it be possible / is it worth it to use some inline functions to reduce the multi-layered flag setting and checking in this section of code?

Reply:
let me look. I think also if the filtering happened through a list of filter lambdas all separately defined could make this clearer.